lib/backport.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Backports to 4.x releases.
+ * Maps:
+ * lines -> first gurmukhi key: gurmukhi
+ * drop current sources.json
+ * compositions.json -> sources.json
+ */
+
+/* eslint-disable global-require, import/no-dynamic-require */
+
+const { unlink, rename, readdir, readFile, writeFile, stat } = require( 'fs/promises' )
+const { reduce, map, forEach } = require( 'fast.js' )
+const { resolve } = require('path')
+
+const colors = require( './string-colors' )
+
+const JSON_PATH = resolve( __dirname, '../data' )
+
+const GURMUKHI_SOURCES = {
+  'Ardaas': 'SGPC',
+  'Ganj Nama Bhai Nand Lal Ji': 'Dr. Ganda Singh',
+  'Ghazals Bhai Nand Lal Ji': 'Dr. Ganda Singh',
+  'Jot Bigas Bhai Nand Lal Ji': 'Dr. Ganda Singh',
+  'Kabit Savaiye Bhai Gurdas Ji': 'Seva Singh',
+  'Sarabloh Granth': 'Amrit Keertan',
+  'Sri Dasam Granth': 'SGPC',
```

Comment:
```suggestion
  'Sri Dasam Granth': 'SGPC' || 'Budha Dal Mehron',
```

SGPC doesn't span the whole composition, so a fallback source is needed. Not sure if this is the proper way to do it.

Reply:
Very good call out, this would not be the way to do it, but I'll update my PR

Reply:
Are there any other cases of this?

Reply:
Let me check

Reply:
LGTM